### ISSUE TITLE
Output format error if single quote used in feature definition

### ISSUE THREAD

**Issue by GregoryBevan** [rmpestano/cukedoctor#170]
Hello,

I know it's probably not a common use case in English, but in French we often need to use single quotes as separator between words. So it's difficult to avoid them in feature definition and I encountered a problem with the documentation generation with the following example :

```gherkin
# language: fr
Fonctionnalité: Parcours de recherche d'un article

  Scénario: Recherche d'un article connu
	  Etant donné que je connais déjà l'article
	  Lorsque je le cherche dans les rayons
	  Alors je le trouve à sa place

```

The single quote in _d'un_ is causing a mess during generation. And the resulting html report could be seen here :
[https://gregorybevan.github.io/sbt-with-cucumber/cucumber.html](https://gregorybevan.github.io/sbt-with-cucumber/cucumber.html)

It seems that the problem comes from the string _Parcours-de-recherche-d'un-article_ in the generated _.adoc_  
[https://gregorybevan.github.io/sbt-with-cucumber/cucumber.adoc](https://gregorybevan.github.io/sbt-with-cucumber/cucumber.adoc)

I have already checked that it's not linked to the language of the feature file.

Sample project available [here](https://github.com/GregoryBevan/sbt-with-cucumber/tree/cukedoctor-feature-bug)

**Comment by rmpestano** [comment]
Hi @GregoryBevan, I've played a bit with your sample project (thanks for that) but couldn't even make it work with pure Asciidoctor.

The main issue with single quotes is on the anchor link:

```
[[Parcours-de-recherche-d'un-article, Parcours de recherche d'un article]]
=== *Parcours de recherche d'un article*
``` 

Asciidoctor use single quotes to enable variable substitutions in links/cross references but even escaping it with `\'` didn't make the trick, still breaks the rendering:

<img width="819" alt="Screenshot 2020-11-22 at 19 08 35" src="https://user-images.githubusercontent.com/1592273/99911651-33594f00-2cf6-11eb-8185-36f1affac02a.png">

On command line I get a warn:

```
AMSrpestano-C02Z409YLVCG:test rpestano$ asciidoctor test.adoc 
asciidoctor: WARNING: test.adoc: line 59: section title out of sequence: expected level 2, got level 3
```

If someone knows how to make cross references work with single quotes, following is the adoc file I'm using:

```
:toc: right
:backend: html5
:doctitle: Doc Title
:doctype: book
:icons: font
:!numbered:
:!linkcss:
:sectanchors:
:sectlink:
:docinfo:
:source-highlighter: highlightjs
:toclevels: 3
:hardbreaks:
:chapter-label: Chapter
:version-label: Version

= *Doc Title*

== *Summary*
[cols="12*^m", options="header,footer"]
|===
3+|Scenarios 7+|Steps 2+|Features: 1

|[green]#*Passed*#
|[red]#*Failed*#
|Total
|[green]#*Passed*#
|[red]#*Failed*#
|[purple]#*Skipped*#
|[maroon]#*Pending*#
|[yellow]#*Undefined*#
|[blue]#*Missing*#
|Total
|Duration
|Status

12+^|*<<Parcours-de-recherche-d\'un-article>>*
|1
|0
|1
|3
|0
|0
|0
|0
|0
|3
|009ms
|[green]#*passed*#
12+^|*Totals*
|1|0|1|3|0|0|0|0|0|3 2+|009ms
|===

== *Features*

[[Parcours-de-recherche-d\'un-article, Parcours de recherche d\'un article]]
=== *Parcours de recherche d\'un article*

==== Scenario: Recherche dune article connu

==========
Given ::
je connais déjà l'article icon:thumbs-up[role="green",title="Passed"] [small right]#(009ms)#
When ::
je le cherche dans les rayons icon:thumbs-up[role="green",title="Passed"] [small right]#(000ms)#
Then ::
je le trouve à sa place icon:thumbs-up[role="green",title="Passed"] [small right]#(000ms)#
==========

```

**Comment by rdlopes** [comment]
No need to add the single quote to the id.
In com.github.cukedoctor.renderer.CukedoctorFeatureRenderer#renderFeatureSectionId the reserved characters for asciidoctor could rather be escaped.

**Comment by rmpestano** [comment]
Hey @rdlopes, that's amazing, thank you!

**Comment by rmpestano** [comment]
This feature is available in `v3.8.0`, thank you guys for the help and making it possible!
